### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by Bigizi**
**OOP school library project**

- Set up associations between classes and objects.
- Run the project using the command line

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -18,4 +22,12 @@ def can_use_services?
...
   end
   public :can_use_services?
+
+  def correct_name
+    @name
+  end
+
+  def add_rental(date, book)
+    @rentals.push(Rental.new(date, book))
+  end
```
Comment (thecodechaser) [inline person.rb]: - [ ] Rental class constructor also expects `person` as a parameter so please make sure to pass the `person` parameter in the Rental class constructor. You can consider the below code.

```ruby
  def add_rental(date, book)
    @rentals.push(Rental.new(date,self, book))
  end

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(date, person)
+    @rentals.push(Rental.new(date, person))
+  end
+end
```
Comment (thecodechaser) [inline book.rb]: - [ ] Rental class constructor also expects `book` as a parameter so please make sure to pass the `book` parameter in the Rental class constructor. You can consider the below code.

```ruby
  def add_rental(date, person)
    @rentals.push(Rental.new(date,person,self))
  end

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Bigizi👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Highly organized code. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️

- [ ] https://github.com/Bigizi/OOP-school-library/blob/ee30329406ff8e87914231260d0aded1e5de06a6/student.rb#L1-L15
  - Please make sure when setting the classroom for a student it also adds it to the classroom's students that serves the 'belongs-to' side (a student belongs to a classroom). You can consider adding the below code in the student class.
```ruby
 def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

- [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AnselemOdims — APPROVED** [review]
### Hi  @Bigizi 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**